File: 17-04-2024/Count Pairs in an Array.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG
{
	public static void main(String[] args) throws IOException
	{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while(t-->0)
        {
            int n = Integer.parseInt(br.readLine().trim());
            int a[] = new int[(int)n];
            String inputLine[] = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                a[i] = Integer.parseInt(inputLine[i]);
            }
            
            Solution obj = new Solution();  
            System.out.println(obj.countPairs(a, n));
            
        }
	}
}

// } Driver Code Ends


//User function Template for Java


//Your Code start

class Solution {
    static int countPairs(int arr[], int n) {
        if (n <= 2) {
            return 0;
        }
        List<Integer> res = new ArrayList<>();
        Map<Integer, Integer> mp = new HashMap<>();
        int count = 0;
        for (int i = 0; i < n; i++) {
            res.add(i * arr[i]);
        }
        List<Integer> dup = new ArrayList<>(res);
        dup.sort(null);

        for (int i = 0; i < n; i++) {
            int index = dup.indexOf(res.get(i));
            if (res.get(i) == 0) {
                dup.remove(index);
                continue;
            }
            dup.remove(index);
            count += index;
        }
        return count;
    }
}
//Your Code End

File: 17-04-2024/Count Pairs in an Array.py
#User function Template for python3
#Your Code Start
class Solution:
    @staticmethod
    def countPairs(arr, n):
        if n <= 2:
            return 0
        res = []
        mp = {}
        count = 0
        for i in range(n):
            res.append(i * arr[i])
        dup = res[:]
        dup.sort()

        for i in range(n):
            index = dup.index(res[i])
            if res[i] == 0:
                dup.pop(index)
                continue
            dup.pop(index)
            count += index
        return count
        #Your Code End

#{ 
 # Driver Code Starts
#Initial Template for Python 3

def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        ob= Solution()
        print(ob.countPairs(a, n))

        T -= 1


if __name__ == "__main__":
    main()
    
# } Driver Code Ends
